activemodel/lib/active_model/attribute_assignment.rb

```diff
@@ -27,7 +27,7 @@ module AttributeAssignment
     #   cat.status # => 'sleeping'
     def assign_attributes(new_attributes)
       if !new_attributes.respond_to?(:stringify_keys)
-        raise ArgumentError, "When assigning attributes, you must pass a hash as an argument."
+        raise ArgumentError, "When assigning attributes, you must pass a hash as an argument, #{new_attributes.class} passed."
```

Comment:
I wonder if we should say "hash-like object" instead of simply "hash"?

Reply:
I think "hash-like object" would be confusing to beginners. Hash is fine.